dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/LoginPage.java

```diff
@@ -184,7 +185,11 @@ public String login() {
             }
 
             logger.log(Level.FINE, "Sending user to = {0}", redirectPage);
-            return redirectPage + (!redirectPage.contains("?") ? "?" : "&") + "faces-redirect=true";
+            if(redirectPage.startsWith("http")) {
```

Comment:
One thing about such redirect. I think that it could cause somewhat of a security threat. Redirect page can be specified by anyone. Let's say that we have malicious user that shares the link with someone. This link have set redirect page to `http://fake-dataverse.com` that looks exactly the same as the original dataverse. User sees exactly the same page but for some reason it is not logged in. It is confusing for him, but he decides to try to log in again, but now the credentials will go to `fake-dataverse.com` and the password is compromised.

Reply:
Improved as proposed